server/src/collaborations/interfaces/ICollaborationsRepository.ts

```diff
@@ -0,0 +1,12 @@
+import { Collaboration } from '@collaborations/infra/typeorm/entities/collaboration.entity';
+import { CreateCollaborationDto } from '@collaborations/dto/create-collaboration.dto';
+import { FilterCollaborationByStatusDto } from '@collaborations/dto/filter-collaboration-by-status.dto';
+
+export abstract class ICollaborationsRepository {
+  abstract create(data: CreateCollaborationDto): Promise<Collaboration>;
+
+  abstract filterCollaborationsByStatus({
```

Comment:
Se esse método for específicamente para colaborações que envolvem o academy, escopar melhor o método e a lógica para trazer apenas as do academy, ou se for pra deixar para todas as colaborações não utilizar ids